<a href="https://colab.research.google.com/github/kaki-rufus/Customer-Analysis-using-KMeans-Algorithm/blob/main/Copy_of_SIT742Task2_checkpoint.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

  Part 1

  Open the assignment2data.json file and convert it to csv format as dataframe in pandas. Removing
the duplicated rows from dataframe and save as the new dataframe. The meaning of the column is in
assignment2data.pdf
• Write a code to change the value of ’Y’ from column is_newsletter_subscriber to 1 and ’N’
to 0. (1 and 0 is ”int” type)
• Print out the value count for column is_newsletter_subscriber.


In [ ]:
# import the necessary dependencies
import json
from google.colab import files
uploaded = files.upload()

Saving assignment2data.json to assignment2data (1).json


In [ ]:
# import pandas library
import pandas as pd

# read the json data
df = pd.read_json("assignment2data.json", lines=True)
# df.head()

# convert the datafareme to csv file and load it
df.to_csv("assignment2data.csv", index=False)

new_df = pd.read_csv("assignment2data.csv")

# remove duplicates
new_df = new_df.drop_duplicates()

In [ ]:
# confirm if there are duplicates
new_df.duplicated().sum()

0

 Answer 1.6

In [ ]:
# map Initial values of Y and N to 1 and 0 respectively
new_df['is_newsletter_subscriber'] = new_df['is_newsletter_subscriber'].map({'Y':1, 'N':0})

# value count for the column (is_newsletter_subscriber)
count = new_df['is_newsletter_subscriber'].value_counts()
print(count)

0    27203
1    18827
Name: is_newsletter_subscriber, dtype: int64


Answer 1.7

In [ ]:
# new_df.columns

In [ ]:
# creating new features
new_df['female_item_rate'] = new_df['female_items'] / new_df['items']
new_df['male_item_rate'] = new_df['male_items'] / new_df['items']
new_df['unisex_items_rate'] = new_df['unisex_items'] / new_df['items']

# rows (customers) having the value (female_item_rate == 1) and the value (male_item_rate == 1) and the value (orders > 4.11)
num_customers = new_df.loc[(new_df['female_item_rate']==1) & (new_df['male_item_rate']==1) & (new_df['orders']>4.11)]
num_customers.shape[0]

0

**QUESTION 1.6**


















Map method was used since it is simple to use with a built in input format, 


DICTIONARY.






























Other methods which can be used in place of Map include one hot encoding, target encoding and label encoders. The latter methods could consume alot time writing . Since we only had 2 unique values to format Map method was prefered






















**VALUE** **COUNTS**:

In this solution, the pandas Value_counts was used to count the number of unique values found in the column. Other methods which could be used are the groupby function which is a long method and time consuming. Value_counts does this in a simple and understandable line of code







0 - 27203






1 - 18827




**QUESTION 1.7**













loc pandas method was used in order to locate the exact labels from the dataframe when setting the condition of the number of customers.
From the conditions, none (0) of the customers had female and male item rate equal to 1 and order greater than 4.11



Shape[0] method was used in order to give the number of rows(customers)